File: app/src/main/java/denis/easyweather/app/model/CityModel.java
package denis.easyweather.app.model;

import java.util.List;

public class CityModel {
    public Coordinates coord;
    public SunData sys;
    public List<WeatherDescription> weather;
    public WeatherData main;
    public WindData wind;
    public RainData rain;
    public Clouds clouds;
    public Long dt;
    public Long id;
    public String name;
    public Integer cod;

    public class WeatherData {
        public Double temp;
        public Double humidity;
        public Double pressure;
        public Double temp_min;
        public Double temp_max;
    }

    public class Coordinates {
        public Double lat;
        public Double lon;
    }

    public class SunData {
        public String country;
        public Long sunrise;
        public Long sunset;
    }

    public class WeatherDescription {
        public Long id;
        public String main;
        public String description;
        public String icon;
    }

    public class WindData {
        public Double speed;
        public Double deg;
    }

    public class RainData {
        public Integer _3h;
    }

    public class Clouds {
        public Integer all;
    }
}

File: app/src/main/java/denis/easyweather/app/model/CityModel.kt
package denis.easyweather.app.model

class CityModel {
    var coord: Coordinates? = null
    var sys: SunData? = null
    var weather: List<WeatherDescription>? = null
    var main: WeatherData? = null
    var wind: WindData? = null
    var rain: RainData? = null
    var clouds: Clouds? = null
    var dt: Long? = null
    var id: Long? = null
    var name: String? = null
    var cod: Int? = null

    inner class WeatherData {
        var temp: Double? = null
        var humidity: Double? = null
        var pressure: Double? = null
        var temp_min: Double? = null
        var temp_max: Double? = null
    }

    inner class Coordinates {
        var lat: Double? = null
        var lon: Double? = null
    }

    inner class SunData {
        var country: String? = null
        var sunrise: Long? = null
        var sunset: Long? = null
    }

    inner class WeatherDescription {
        var id: Long? = null
        var main: String? = null
        var description: String? = null
        var icon: String? = null
    }

    inner class WindData {
        var speed: Double? = null
        var deg: Double? = null
    }

    inner class RainData {
        var _3h: Int? = null
    }

    inner class Clouds {
        var all: Int? = null
    }
}
